### PR TITLE
Document apt-get workaround

### PR DESCRIPTION
**Pull request by a3nm**
Hi, this is as discussed in issue #1. I also documented the need to install the dependencies of vmdb2, if that's OK.

By the way, one other thing that could maybe be documented is the fact that building the image fails mysteriously ("cannot run apt-key") if the user running the build has another umask that 022 (see [here](https://github.com/larswirzenius/vmdb2/issues/26#issuecomment-339157527)). I don't know whether you think it's a relevant point to document, but it's the main problem I had to debug when building the image. What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,11 +9,35 @@ See https://wiki.debian.org/RaspberryPi3#Preview_image for where to obtain the l
...
+- If you want to use the snapshot with which the build was tested, use
+    `http://snapshot.debian.org/archive/debian/20171007T213914Z/`. This is what
+    is pre-configured in raspi3.yaml. However, due to a [missing
+    feature](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=763419) on
+    snapshots, to make the build work, you have to disable an expiration check
+    by APT. To do so, edit raspi3.yaml to replace all
+    `apt-get` invocations with `apt-get -o Acquire::Check-Valid-Until=false`
+- If you want to use the latest versions of each software, you can replace
+    `http://snapshot.debian.org/archive/debian/20171007T213914Z/` in raspi3.yaml
+    with the URL of your favorite Debian mirror. Of course, this means that the
```
Comment (stapelberg) [inline README.md]: s,the URL of your favorite Debian mirror,http://deb.debian.org/debian, please. People who care about mirrors will know how to replace it.

**Review by stapelberg — CHANGES_REQUESTED** [review]

**Comment by stapelberg** [comment]
Thanks for the PR!

> By the way, one other thing that could maybe be documented is the fact that building the image fails mysteriously ("cannot run apt-key") if the user running the build has another umask that 022 (see here). I don't know whether you think it's a relevant point to document, but it's the main problem I had to debug when building the image. What do you think?

Yeah, we should definitely recommend `env -i` and `umask` until https://github.com/larswirzenius/vmdb2/issues/26 is fixed upstream. Feel free to put that into this very PR, or submit a different one.

File: README.md (hunk 1 above)
Comment (a3nm) [reply README.md]: OK great idea! fixing

**Comment by a3nm** [comment]
I implemented your suggestion, thanks! As for the environment cleanup issue, I tried to explain how to do it, but with `env -i` it's a bit ugly because you need to keep some variables. What do you think?

I'm running a build now to check that the proposed invocation works until the end; if it does work I'm fine with merging.

File: README.md [hunk 2]
```diff
@@ -9,12 +9,39 @@ See https://wiki.debian.org/RaspberryPi3#Preview_image for where to obtain the l
...
+    with `http://deb.debian.org/debian`. Of course, this means that the
+    build may break or fail to boot if there are regressions in the latest
+    versions.
+
+Once you have edited raspi3.yaml, you can generate the image by
+issuing the following:
+
+```shell
+umask 022
+sudo env -i HOME="$HOME" LC_CTYPE="${LC_ALL:-${LC_CTYPE:-$LANG}}" PATH="$PATH" USER="$USER" \
```
Comment (stapelberg) [inline README.md]: Do you know why $HOME and $USER are required? This strikes me as odd.

We should set `LC_CTYPE=C.UTF-8`.

Comment (a3nm) [reply README.md]: $HOME and $USER were probably not required, I just lost patience after $LT_CTYPE and $PATH so I searched for a solution online. Fixed.

**Comment by a3nm** [comment]
I did your suggested change, thanks! The build seems to run, just checking now that it actually completes.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Please let me know once you have a working image and I’ll merge. Thanks again!

**Comment by a3nm** [comment]
Image creation completed successfully, so for me this can be merged. Thanks for reviewing and for coding this. :)

-- 
Antoine Amarilli

**Pull request merged**